gh-112139: Add `inspect.Signature.format` and use it in `pydoc` (#112143)
python/cpython: Doc/library/inspect.rst

      Signature objects are also supported by generic function
      :func:`copy.replace`.

   .. method:: format(*, max_width=None)

      Convert signature object to string.

>       If *max_width* integer is passed,
>       signature will try to fit into the *max_width*.
>       If signature is longer than *max_width*,
>       all parameters will be on separate lines.

Reviewer: ```suggestion
      If *max_width* is passed, the method will attempt to fit
      the signature into lines of at most *max_width* characters.
      If the signature is longer than *max_width*,
      all parameters will be on separate lines.
```
